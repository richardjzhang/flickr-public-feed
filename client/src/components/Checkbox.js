// @flow strict
/** @jsx jsx */
import { jsx } from '@emotion/core';
import styled from '@emotion/styled';
import { HStack, HSpacer } from 'src/components/Stack';
import { BASE_UNIT, colors, fontSize, fontWeight } from 'src/styles';

const CheckboxStyle = styled.input<{}>({
  borderRadius: BASE_UNIT,
  backgroundColor: colors.white,
  margin: 0,
});

const Text = styled.div<{}>({
  color: colors.grey10,
  fontSize: fontSize.small,
  fontWeight: fontWeight.semibold,
});

export default function Checkbox({
  value,
  onChange,
  text,
}: {|
  value: boolean,
  onChange: () => void,
  text: string,
|}) {
  return (
    <HStack css={{ alignItems: 'center' }}>
      <CheckboxStyle type="checkbox" value={value} onChange={onChange} />
      <HSpacer width={BASE_UNIT} />
      <Text>{text}</Text>
    </HStack>
  );
}
