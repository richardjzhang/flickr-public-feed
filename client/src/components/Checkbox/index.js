// @flow strict
/** @jsx jsx */
import { jsx } from '@emotion/core';
import styled from '@emotion/styled';
import { HStack, HSpacer } from 'src/components/Stack';
import { BASE_UNIT, colors, fontSize, fontWeight } from 'src/styles';

const CheckboxStyle = styled.input<{}>({
  height: 4 * BASE_UNIT,
  width: 4 * BASE_UNIT,
  borderRadius: BASE_UNIT,
  backgroundColor: colors.white,
  margin: 0,
  cursor: 'pointer',
});

const Text = styled.div<{}>({
  color: colors.grey,
  fontSize: fontSize.medium,
  fontWeight: fontWeight.semibold,
});

export default function Checkbox({
  value,
  onChange,
  text,
}: {|
  value: boolean,
  onChange: () => void,
  text: string,
|}) {
  return (
    <HStack
      css={{
        alignItems: 'center',
        padding: `${2 * BASE_UNIT}px ${4 * BASE_UNIT}px`,
        width: 'fit-content',
        backgroundColor: colors.white,
        borderRadius: BASE_UNIT,
      }}
    >
      <CheckboxStyle type="checkbox" checked={value} onChange={onChange} />
      <HSpacer width={2 * BASE_UNIT} />
      <Text>{text}</Text>
    </HStack>
  );
}
