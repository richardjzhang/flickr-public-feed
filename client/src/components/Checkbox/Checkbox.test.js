import { render } from '@testing-library/react';
import * as React from 'react';

import Checkbox from '.';

describe('Checkbox', () => {
  test('renders correctly with value of false', () => {
    const tree = render(
      <Checkbox value={false} onChange={() => {}} text="Checkbox text" />,
    );
    expect(tree).toMatchSnapshot();

    const checkboxText = tree.queryByText('Checkbox text');
    expect(checkboxText).toBeInTheDocument();
  });

  test('renders correctly with value of true', () => {
    const tree = render(
      <Checkbox value={true} onChange={() => {}} text="Checkbox text" />,
    );
    expect(tree).toMatchSnapshot();

    const checkboxText = tree.queryByText('Checkbox text');
    expect(checkboxText).toBeInTheDocument();
  });
});
