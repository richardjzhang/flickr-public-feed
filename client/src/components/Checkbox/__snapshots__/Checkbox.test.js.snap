// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly with value of false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1crab1l-Checkbox"
      >
        <input
          class="css-1xr1lj7"
          type="checkbox"
        />
        <div
          class="css-1rnx2nk-HSpacer"
        />
        <div
          class="css-u4h7m5"
        >
          Checkbox text
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1crab1l-Checkbox"
    >
      <input
        class="css-1xr1lj7"
        type="checkbox"
      />
      <div
        class="css-1rnx2nk-HSpacer"
      />
      <div
        class="css-u4h7m5"
      >
        Checkbox text
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Checkbox renders correctly with value of true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1crab1l-Checkbox"
      >
        <input
          checked=""
          class="css-1xr1lj7"
          type="checkbox"
        />
        <div
          class="css-1rnx2nk-HSpacer"
        />
        <div
          class="css-u4h7m5"
        >
          Checkbox text
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1crab1l-Checkbox"
    >
      <input
        checked=""
        class="css-1xr1lj7"
        type="checkbox"
      />
      <div
        class="css-1rnx2nk-HSpacer"
      />
      <div
        class="css-u4h7m5"
      >
        Checkbox text
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
